### PR TITLE
[GITHUB-17] Update to Latest Standards

### PR DESCRIPTION
**Pull request by dennisconrad**
* Flatten variables
* Migrate to molecule tests
* Closes (previously fixed) #15

### CODE REVIEW

File: molecule/default/playbook.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      apt:
+        name: "{{ item }}"
+        update_cache: yes
+        cache_valid_time: 3600
+      with_items:
+        - apt-transport-https
+        - net-tools
+
+  roles:
+    - role: sansible.zookeeper
```
Comment (lobsterdore) [inline molecule/default/playbook.yml]: New line here.

File: meta/main.yml [hunk 2]
```diff
@@ -1,22 +1,24 @@
...
-      users:
-        - name: kafka
-          group: kafka
+    sansible_users_and_groups_groups:
+      - name: "{{ sansible_kafka_group }}"
+    sansible_users_and_groups_users:
+      - name: "{{ sansible_kafka_user }}"
+        group: "{{ sansible_kafka_group }}"
+
+  - role: sansible.zookeeper
```
Comment (lobsterdore) [inline meta/main.yml]: Zookeeper should only be included for tests, it will live on a separate machine in the real world.

File: meta/main.yml [hunk 3]
```diff
@@ -1,22 +1,24 @@
...
   platforms:
     - name: Ubuntu
       versions:
         - trusty
+        - xenial
+        - bionic
   categories:
     - development
 
 dependencies:
```
Comment (lobsterdore) [inline meta/main.yml]: We need sansible.java as a dep here.

File: defaults/main.yml [hunk 4]
```diff
@@ -1,60 +1,50 @@
...
-  # Note that you must install the AWS CLI tools on the machine to use this feature
-  aws_cluster_autodiscover:
-    enabled: false
-    hosts: []
-    # Tag to store the ID/index of the host that gets assigned to a machine
-    id_tag_name: KafkaID
-    # AWS CLI lookup filter
-    lookup_filter: "Name=tag:Environment,Values=dev Name=tag:Role,Values=kafka"
-    r53_zone_id: ~
+sansible_kafka_apache_mirror: https://archive.apache.org/dist/
```
Comment (lobsterdore) [inline defaults/main.yml]: Probably seem some vars for java version and vendor (similar to your ZK changes).

File: meta/main.yml (hunk 2 above)
Comment (dennisconrad) [reply meta/main.yml]: That wasn't included because zookeeper had that dependency.

File: meta/main.yml (hunk 3 above)
Comment (dennisconrad) [reply meta/main.yml]: Added.

File: defaults/main.yml (hunk 4 above)
Comment (dennisconrad) [reply defaults/main.yml]: Added.

**Comment by lobsterdore** [comment]
LGTM 👍

**Pull request merged**